### ISSUE TITLE
Serialized data interface

### ISSUE THREAD

**Issue by jmadden173** [jlab-sensing/soil-power-sensor-firmware#6]
Serialized measurement data will be used to store data on the FRAM chip and reduce packet size for LoRaWAN uploads. Needs to be determined if necessary for uploading via WiFi. Serialization will happen with protobuf.

**Comment by jmadden173** [comment]
Tried implementing `nanopb` through the platformio registry and got the following

```
...
In file included from .pio/build/stm32/nanopb/generated-src/soil_power_sensor.pb.c:4:0:
.pio/build/stm32/nanopb/generated-src/soil_power_sensor.pb.h:7:10: fatal error: google/protobuf/timestamp.pb.h: No such file or directory
 #include "google/protobuf/timestamp.pb.h"
```

**Comment by jmadden173** [comment]
Also I'm at an impass for implementing protobuf in this project. Here are the following solutions I have considered

1. Use standard protobuf with generated c++ code
    - Don’t like the idea of the added bulk of using C++ folder
2. Use protobuf-c in a separate folder
    - Would need to create a separate library that will be imported to the stm32 project
    - Needs the library to be compiled on the host device
3. Use nanopb in a separate folder
    - Same issues with above
4. use nanopb with platformio implementation
    - Does not seem to support well known types listed on the wiki aka timestamp
    - How do I then create something like a python package that can be used later by Dirtviz

**Comment by jmadden173** [comment]
Accidentally closed as part of #17
